Framework/Kernel/src/GitHubApiHelper.cpp

```diff
@@ -0,0 +1,120 @@
+#include "MantidKernel/GitHubApiHelper.h"
+#include "MantidKernel/DateAndTime.h"
+#include "MantidKernel/Logger.h"
+#include <Poco/StreamCopier.h>
+#include <Poco/URI.h>
+#include <Poco/Net/HTTPRequest.h>
+#include <Poco/Net/HTTPResponse.h>
+#include <Poco/Net/HTTPSClientSession.h>
+
+namespace Mantid {
+namespace Kernel {
+
+using namespace Poco::Net;
+using std::map;
+using std::string;
+
+namespace {
+// anonymous namespace for some utility functions
+/// static Logger object
+Logger g_log("InternetHelper");
+}
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+*/
+GitHubApiHelper::GitHubApiHelper() : InternetHelper() {
+  addAuthenticationToken();
+}
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+*/
+GitHubApiHelper::GitHubApiHelper(const Kernel::ProxyInfo &proxy)
+    : InternetHelper(proxy) {
+  addAuthenticationToken();
+}
+
+//----------------------------------------------------------------------------------------------
+/** Destructor
+*/
+GitHubApiHelper::~GitHubApiHelper() {}
+
+void GitHubApiHelper::reset() {
+  InternetHelper::reset();
+  addAuthenticationToken();
+}
+
+bool GitHubApiHelper::isAuthenticated() {
+  return (m_headers.find("Authorization") != m_headers.end());
+}
+
+void GitHubApiHelper::processResponseHeaders(
+    const Poco::Net::HTTPResponse &res) {
+  // get github api rate limit information if available;
+  int rateLimitRemaining = 0;
+  int rateLimitLimit;
+  DateAndTime rateLimitReset;
+  try {
+    rateLimitLimit =
+        boost::lexical_cast<int>(res.get("X-RateLimit-Limit", "-1"));
+    rateLimitRemaining =
+        boost::lexical_cast<int>(res.get("X-RateLimit-Remaining", "-1"));
+    rateLimitReset.set_from_time_t(
+        boost::lexical_cast<int>(res.get("X-RateLimit-Reset", "0")));
+  } catch (boost::bad_lexical_cast const &) {
+    rateLimitLimit = -1;
+  }
+  if (rateLimitLimit > -1) {
+    g_log.debug() << "GitHub API " << rateLimitRemaining << " of "
+                  << rateLimitLimit << " calls left. Resets at "
+                  << rateLimitReset.toSimpleString() << " GMT" << std::endl;
```

Comment:
Not that it matters much in this type of message, but for consistency all `std::endl` were replaced by `'\n'` recently, and it would be good to stick to that:   #16563. This line could end with `" GMT\n";`.